### ISSUE TITLE
Support for TanStack Start

### ISSUE THREAD

**Issue by ian-pascoe** [anomalyco/sst#4198]
I'm sick of Next, don't want to use Remix, and have been hearing a lot about TanStack Start. 

Since it is similar to SolidStart, hopefully it's not too hard to implement.

**Comment by ax-at** [comment]
Any update on this? @thdxr

**Comment by matheuslanduci** [comment]
I got it working using `new sst.aws.TanstackStart` (undocumented) but I had to switch the `Invoke mode` to `RESPONSE_STREAM` instead of `BUFFERED` (the default)

![image](https://github.com/user-attachments/assets/84927e8e-4338-446a-b588-0dfe7a216bc1)

It was working when I tried to call the function via CLI/Dashboard but via URL wasn't working

**Comment by mckamyk** [comment]
Same

**Comment by jtirene** [comment]
To piggyback onto @matheuslanduci's suggestion, you can set the `Invoke mode` programmatically like this:
```
// sst.config.ts
new sst.aws.TanstackStart('MyTanstackStart', {
  transform: {
    server: {
      streaming: true,
    },
  },
})
```
[Source](https://github.com/sst/sst/blob/cf6dc58677191bb9a655397810f8ccc0591d335f/platform/src/components/aws/function.ts#L2220)

**Comment by ivawzh** [comment]
Does it support the new version of Tanstack Start after they migrated off devinxi?

**Comment by mckamyk** [comment]
Yes, it does, coincidentally. Its not documented in TSS side, but you'll still want a `nitro.config.ts` with `awsLambda.streaming: true` and `inlineDynamicImports: true`
